tests/SpecTests/TransactionsSpecTest.php

```diff
@@ -0,0 +1,236 @@
+
+        foreach ($expectedCommand as $key => $value) {
+            if ($value === null) {
+                $this->assertObjectNotHasAttribute($key, $actualCommand);
+                unset($expectedCommand[$key]);
+            }
+        }
+
+        $this->assertMatchesDocument($expectedCommand, $actualCommand);
+    }
+
+    /**
+     * Execute an individual test case from the specification.
+     *
+     * @dataProvider provideTests
+     * @param string $name           Test name
+     * @param array  $test           Individual "tests[]" document
+     * @param array  $runOn          Server requirements for the test
+     * @param array  $data           Data fixtures to initialize collection under test
+     * @param string $databaseName   Name of database under test
+     * @param string $collectionName Name of collection under test
+     */
+    public function testTransactions($name, array $test, array $runOn = null, array $data, $databaseName = null, $collectionName = null)
+    {
+        $this->setName($name);
+
+        // TODO: Revise this once a test environment with multiple mongos nodes is available (see: PHPLIB-430)
+        if (isset($test['useMultipleMongoses']) && $test['useMultipleMongoses'] && $this->isShardedCluster()) {
+            $this->markTestSkipped('"useMultipleMongoses" is not supported');
+        }
+
+        if (isset($runOn)) {
+            $this->checkServerRequirements($runOn);
+        }
+
+        if (isset($test['skipReason'])) {
+            $this->markTestSkipped($test['skipReason']);
+        }
+
+        $databaseName = isset($databaseName) ? $databaseName : $this->getDatabaseName();
+        $collectionName = isset($collectionName) ? $collectionName : $this->getCollectionName();
+
+        $context = Context::fromTransactions($test, $databaseName, $collectionName);
+        $this->setContext($context);
+
+        $this->dropTestAndOutcomeCollections();
+        $this->createTestCollection();
+        $this->insertDataFixtures($data);
+        $this->preventStaleDbVersionError($test['operations']);
+
+        if (isset($test['failPoint'])) {
+            $this->configureFailPoint($test['failPoint']);
+        }
+
+        if (isset($test['expectations'])) {
+            $commandExpectations = CommandExpectations::fromTransactions($test['expectations']);
+            $commandExpectations->startMonitoring();
+        }
+
+        foreach ($test['operations'] as $operation) {
+            Operation::fromTransactions($operation)->assert($this, $context);
+        }
+
+        $context->session0->endSession();
+        $context->session1->endSession();
+
+        if (isset($commandExpectations)) {
+            $commandExpectations->stopMonitoring();
+            $commandExpectations->assert($this, $context);
+        }
+
+        if (isset($test['outcome']['collection']['data'])) {
+            $this->assertOutcomeCollectionData($test['outcome']['collection']['data']);
```

Comment:
Does this fulfill "Ensure this find reads the latest data by using primary read preference with local read concern even when the MongoClient is configured with another read preference or read concern."?